mig-loader/loader_test.go

```diff
@@ -0,0 +1,428 @@
+    nMU6XkLw3T3GZPYdGbKcfZ/+P8Yo/DywayzkYiYBQsK50xunGwADfHUcoNNjJij/oPm5lKmEF6fi2U
+    HwX2ugHGcjGtanytPyNJVaWwtf3zOHPK9xkY9JzjsmVqD+lAjRMe63WzpZxp8MIrwWT7G++a54ohD6
+    ficeOKtuRb1yYo3Z2jwPBHyVFXuKwKUDM0xqEK7JuG39Q098d7wF6YsyyAnTEka9d58XiZXM==/JuZ"
+  ]
+}`
+
+var manNoSignature = `
+{
+  "loader_name": "",
+  "entries": [
+    {
+      "name": "mig-loader",
+      "sha256": "355b34f99acaeb1ab18a7e08b7f19d54b7915fb234777bee828ed5e34b30c9de"
+    },
+    {
+      "name": "configuration",
+      "sha256": "6dcf3bc90eb5c76528edcb1f71bed7dccb886463a5c0c19b33bc1e8d25ce82d8"
+    },
+    {
+      "name": "mig-agent",
+      "sha256": "ce7e7588073021d590ede399dcf31f9148da969e333085a3e58ef78f9d8c31f2"
+    },
+    {
+      "name": "agentcert",
+      "sha256": "017525f2f851311e9b0e26a139252c13b186a6507206cbd0dcc1ca35258b9566"
+    },
+    {
+      "name": "cacert",
+      "sha256": "215394a591db4dbf2bbbb17a4d45b5bc6d335d15a7d2c42876d4b27f8269bda9"
+    },
+    {
+      "name": "agentkey",
+      "sha256": "88df8f032916dfa0ae6c4778fd2aa2084c1aac017aab70f7d4bc6f4327c5c24c"
+    }
+  ],
+  "signatures": []
+}`
+
+var manDupSigs = `
+{
+  "loader_name": "",
+  "entries": [
+    {
+      "name": "mig-loader",
+      "sha256": "355b34f99acaeb1ab18a7e08b7f19d54b7915fb234777bee828ed5e34b30c9de"
+    },
+    {
+      "name": "configuration",
+      "sha256": "6dcf3bc90eb5c76528edcb1f71bed7dccb886463a5c0c19b33bc1e8d25ce82d8"
+    },
+    {
+      "name": "mig-agent",
+      "sha256": "ce7e7588073021d590ede399dcf31f9148da969e333085a3e58ef78f9d8c31f2"
+    },
+    {
+      "name": "agentcert",
+      "sha256": "017525f2f851311e9b0e26a139252c13b186a6507206cbd0dcc1ca35258b9566"
+    },
+    {
+      "name": "cacert",
+      "sha256": "215394a591db4dbf2bbbb17a4d45b5bc6d335d15a7d2c42876d4b27f8269bda9"
+    },
+    {
+      "name": "agentkey",
+      "sha256": "88df8f032916dfa0ae6c4778fd2aa2084c1aac017aab70f7d4bc6f4327c5c24c"
+    }
+  ],
+  "signatures": [
+    "wsBcBAABCAAQBQJXthNjCRDaHVHlrMttfwAAxaoIAFq4OSBL0kGGLqVhobmHXqqxLvEbOhdpcT5Ps
+    YMhi+CMzE429mPatqB2PukZrTjg9z2dJgCOreGMk3PdeEG3HttfHZoXKc73jheZiwewSXGkopWxlBN
+    s35t+FqurpGggJTJ9M8MtK7orCxFE/ei3AYRu6vELhUz+0A5EpI2Fwuo5stiGQGxpNQG4QhZIphbZn
+    g7PThb9Y2f1WkwUoyiDRmzjnrDCt9XYXbrywUSfWawWMmBW/Qmq58IQNmxWGK3mZq1/oQJdextS24J
+    1LJeuCR41+EcqsLS9P7ujFyQD3fbKOPvk8krzFdVHpRafevXwaORc6j2hO9vVV/kgzZ2Ym8A==hdtT",
+    "wsBcBAABCAAQBQJXthNjCRDaHVHlrMttfwAAxaoIAFq4OSBL0kGGLqVhobmHXqqxLvEbOhdpcT5Ps
+    YMhi+CMzE429mPatqB2PukZrTjg9z2dJgCOreGMk3PdeEG3HttfHZoXKc73jheZiwewSXGkopWxlBN
+    s35t+FqurpGggJTJ9M8MtK7orCxFE/ei3AYRu6vELhUz+0A5EpI2Fwuo5stiGQGxpNQG4QhZIphbZn
+    g7PThb9Y2f1WkwUoyiDRmzjnrDCt9XYXbrywUSfWawWMmBW/Qmq58IQNmxWGK3mZq1/oQJdextS24J
+    1LJeuCR41+EcqsLS9P7ujFyQD3fbKOPvk8krzFdVHpRafevXwaORc6j2hO9vVV/kgzZ2Ym8A==hdtT"
+  ]
+}`
+
+func bootstrap() {
+	ctx.Channels.Log = make(chan mig.Log, 0)
+	go func() {
+		for {
+			_ = <-ctx.Channels.Log
+		}
+	}()
+}
```

Comment:
checkManifestSignature in the test functions logs, so this just sets up a routine to drain the logging channel so it doesn't block